# Problem 1
Goal: Probability of Covid Given Test Pool Is Positive.

# Introduction

How it works:

N = Number of people in a test pool.

p = Percentage chance an individual has covid given a test was done, a number.

P(A) = Probability an individual has Covid given a Test was done = p.

P(B) = Probability the test pool is postive.

P(A|B) = The probability an individual tests positive for covid given the test pool is positive.

P(B|A) = The probability the test pool is positive given an individual tests positive in the test pool = 1, because if the individual in the test pool tests postive the whole pool will test positive.

P(Ac) = Probability an individual does not have a positive Covid test result.

P(B|Ac) = The probability a test pool is postiive given the individual does not have covid, meaning 1 to N-1 people in the pool have covid but not A.

The goal of this work is to calculate P(A|B), the probability an individual tests positive for covid given the test pool is positive.

Using Bayes Theorem:

$P(A|B)=\frac{P(B|A)P(A)}{P(B|A)P(A)+P(B|Ac)P(Ac)}$

Now, P(B|A) = 1 as previously stated, P(A) = p, and P(Ac) = 1 - P(A) = 1 - p.
Thus, this reduces to:

$P(A|B)=\frac{p}{p+P(B|Ac)(1-p)}$

Now, P(B|Ac) can be calculated using the binomial distribution, which will give us the probability of x sucesses given a probability p, and a number of trials N.

Binomial Probability:
$P(x;p,n)=\frac{n!}{x!(n-x)!}p^x(1-p)^{n-x}$

This equation is embeded in the Tensorflow probability binomial distribution function which was used.  The results were summed for x = 1 to N-1 sucesses.

Now, essentially we can calculate the probability an individual tests positive for Covid given the test pool had a positive result using only p (the probability an individual has covid given a test for covid) and N (the number of people in a test pool).

In [1]:
import tensorflow_probability as tfp
import numpy as np

State N, the number of people in a test pool, and p, the probability of an idividual testing positive for Covid given a test was conducted.

In [2]:
N = 12 # Number of people in a test pool
p = 0.05 # Probability of an individual testing positive for Covid

Create a tensorflow probability binomial distribution.

In [3]:
X = tfp.distributions.Binomial(N, logits=None, probs=p)

Below, I calculate the probability your result is positive given a positive test pool result.

In [4]:
# AB is the probability of a positive test for you given the test pool was positive.
# BAc is the Probability 1 to N-1 people test positive but not you in your test pool.
BAc = 0
for i in range(1,N-1):
    BAc = BAc + X.prob(i)
AB = p/(p+(BAc*(1-p)))

In [5]:
print(AB)

tf.Tensor(0.10274155, shape=(), dtype=float32)


The probabily an individual tests positive for covid given the test pool tested positive for covid is 10.27% with an infection precent of 5% and a test pool of 12 people.

# References

For Bayes Theorem: https://www.statisticshowto.com/probability-and-statistics/probability-main-index/bayes-theorem-problems/

For Binomial Probability: https://www.statisticshowto.com/probability-and-statistics/binomial-theorem/binomial-distribution-formula/

For Tensorflow Probability Binomial Distributions: https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Binomial